akka-cluster-typed/src/test/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionistSpec.scala

```diff
@@ -567,5 +567,173 @@ class ClusterReceptionistSpec extends WordSpec with Matchers with LogCapturing {
+      } finally {
+        testKit1.shutdownTestKit()
+        testKit2.shutdownTestKit()
+      }
+
+    }
+
+    "handle unregistration per key not per actor" in {
+      val testKit1 = ActorTestKit("ClusterReceptionistSpec-test-11", ClusterReceptionistSpec.config)
+      val system1 = testKit1.system
+      val testKit2 = ActorTestKit(system1.name, system1.settings.config)
+      val system2 = testKit2.system
+      try {
+
+        val clusterNode1 = Cluster(system1)
+        clusterNode1.manager ! Join(clusterNode1.selfMember.address)
+        val clusterNode2 = Cluster(system2)
+        clusterNode2.manager ! Join(clusterNode1.selfMember.address)
+
+        val regProbe1 = TestProbe[Any]()(system1)
+        val regProbe2 = TestProbe[Any]()(system2)
+
+        regProbe1.awaitAssert(clusterNode1.state.members.count(_.status == MemberStatus.Up) should ===(2), 10.seconds)
+
+        system2.receptionist ! Subscribe(PingKey, regProbe2.ref)
+        regProbe2.expectMessage(Listing(PingKey, Set.empty[ActorRef[PingProtocol]]))
+        system2.receptionist ! Subscribe(AnotherKey, regProbe2.ref)
+        regProbe2.expectMessage(Listing(AnotherKey, Set.empty[ActorRef[PingProtocol]]))
+
+        // register same actor for two service keys and verify seen on remote side
+        val service1 = testKit1.spawn(pingPongBehavior)
+        system1.receptionist ! Register(PingKey, service1, regProbe1.ref)
+        regProbe1.expectMessage(Registered(PingKey, service1))
+        regProbe2.expectMessageType[Listing].serviceInstances(PingKey).size should ===(1)
+        system1.receptionist ! Register(AnotherKey, service1, regProbe1.ref)
+        regProbe1.expectMessage(Registered(AnotherKey, service1))
+        regProbe2.expectMessageType[Listing].serviceInstances(AnotherKey).size should ===(1)
+
+        // unregister service1 for one of the keys and verify
+        system1.receptionist ! Deregister(PingKey, service1, regProbe1.ref)
+        regProbe1.expectMessage(Deregistered(PingKey, service1))
+        regProbe2.expectMessageType[Listing].serviceInstances(PingKey).size should ===(0)
+        system2.receptionist ! Find(AnotherKey, regProbe2.ref)
+        regProbe2.expectMessageType[Listing].serviceInstances(AnotherKey).size should ===(1)
+
+        system1.receptionist ! Find(PingKey, regProbe1.ref)
+        regProbe1.expectMessageType[Listing].serviceInstances(PingKey).size should ===(0)
+        system1.receptionist ! Find(AnotherKey, regProbe1.ref)
+        regProbe1.expectMessageType[Listing].serviceInstances(AnotherKey).size should ===(1)
+
+        akka.cluster.Cluster(system1).shutdown()
+        clusterNode2.manager ! Down(clusterNode1.selfMember.address)
+      } finally {
+        testKit1.shutdownTestKit()
+        testKit2.shutdownTestKit()
+      }
+
+    }
+
+    "handle concurrent unregistration and registration on different nodes" in {
+      // this covers the fact that with ddata a removal can be lost
```

Comment:
good, does this test fail (sometimes) if you remove/disable the tombstone stuff?